src/components/frontend-engine/yup/helper.ts

```diff
@@ -12,8 +12,9 @@ import {
 
 interface IYupCombinedRule extends IYupRenderRule, IYupValidationRule {}
 
-// TODO: custom validation
 export namespace YupHelper {
+	const customYupConditions: string[] = [];
```

Comment:
this is static, so if multiple form engines are rendered this could potentially (1) keep expanding and have duplicates or (2) allow a form engine to access a custom rule it didn't define itself. for (1) a Set would resolve it, for (2) I'm thinking is not a problem because the `IYupValidationRule` typing would restrict this

Reply:
I cant see a sensible way out of this actually, at its core, it is relying on `Yup.addMethod()` which is also a static function. even if we deconflict the conditions internally, multiple instances of the forms will still create the rules in the same Yup class.

Reply:
this may be a problem if user is rendering two forms simultaneously AND adding custom validators with the same name but different logic. but I do feel it's unlikely.

then I think minimally we can dedupe `customYupConditions` using a set

was wondering if it's possible to use `schema.test()` instead, but it seems kind of complicated to use.

Reply:
yeah I get what you mean, with `Set` I can dedupe the condition name but there is still an ambiguity in whether it is applied. in view of that, for clarity's sake I added a check against `customYupConditions` to see if the condition is applied already, if yes, it will not be applied again and does a `console.error`